crates/core-subsystem/core-resolver/src/validation/arguments_validator.rs

```diff
@@ -411,3 +423,63 @@ impl<'a> ArgumentValidator<'a> {
         }
     }
 }
+
+fn get_schema_directives(schema: &Schema, base_type_name: &str) -> Vec<ConstDirective> {
+    schema
+        .type_definitions
+        .iter()
+        .filter(|td| matches!(&td.kind, TypeKind::Object(_)))
+        .flat_map(|td| {
+            td.fields()
+                .unwrap()
```

Comment:
most likely also this should be improved

Reply:
Yes, you can make this return `Result<Vec<ConstDirective>, ValidationError>`. You may need to introduce a variation in `ValidationError` (something like `InvalidValidationInfo`).